src/ProtobufServer/ecdar_requests/send_query.rs

```diff
@@ -72,6 +88,50 @@ fn parse_query(query_request: &QueryRequest) -> Result<Query, Status> {
     }
 }
 
+fn parse_components_if_some(
+    proto_component: &ProtobufComponent,
+) -> Result<Vec<Component>, tonic::Status> {
+    if let Some(rep) = &proto_component.rep {
+        match rep {
+            Rep::Json(json) => parse_json_component(json),
+            Rep::Xml(xml) => Ok(parse_xml_components(xml)),
+        }
+    } else {
+        Ok(vec![])
+    }
+}
+
+fn parse_json_component(json: &str) -> Result<Vec<Component>, tonic::Status> {
+    match json_to_component(json) {
+        Ok(comp) => Ok(vec![comp]),
+        Err(_) => Err(tonic::Status::invalid_argument(
+            "Failed to parse json component",
+        )),
+    }
+}
+
+fn parse_xml_components(xml: &str) -> Vec<Component> {
+    let (comps, _, _) = parse_xml_from_str(xml);
+    comps
+}
```

Comment:
This is a bit confusing, but I understand why it turned out so. I would be great if a parse_xml_component existed or vice versa...